components/core/src/clp/streaming_archive/ArchiveMetadata.cpp

```diff
@@ -22,15 +26,24 @@ ArchiveMetadata::ArchiveMetadata(
                          + sizeof(m_end_timestamp) + sizeof(m_compressed_size);
 }
 
-ArchiveMetadata::ArchiveMetadata(FileReader& file_reader) {
-    file_reader.read_numeric_value(m_archive_format_version, false);
-    file_reader.read_numeric_value(m_creator_id_len, false);
-    file_reader.read_string(m_creator_id_len, m_creator_id, false);
-    file_reader.read_numeric_value(m_creation_idx, false);
-    file_reader.read_numeric_value(m_uncompressed_size, false);
-    file_reader.read_numeric_value(m_compressed_size, false);
-    file_reader.read_numeric_value(m_begin_timestamp, false);
-    file_reader.read_numeric_value(m_end_timestamp, false);
+auto ArchiveMetadata::create_from_file_reader(FileReader& file_reader) -> ArchiveMetadata {
+    struct stat file_stat{};
+    auto clp_rc = file_reader.try_fstat(file_stat);
+    if (clp::ErrorCode::ErrorCode_Success != clp_rc) {
+        throw OperationFailed(ErrorCode_Failure, __FILENAME__, __LINE__);
+    }
```

Comment:
another way to write this code is:
Do we prefer one way to another? @LinZhihao-723 

```suggestion
    auto clp_rc = file_reader.try_fstat(file_stat);
    if (auto const clp_rc = file_reader.try_fstat(file_stat);
        clp::ErrorCode::ErrorCode_Success != clp_rc) {
        throw OperationFailed(ErrorCode_Failure, __FILENAME__, __LINE__);
    }
```